docker/update_or_build.sh

```diff
@@ -44,18 +44,38 @@ GIT_REV=$(git rev-parse --short=8 HEAD)
 
 [ "$BRANCH" != "" ] || { echo "-b branch must be set"; usage; exit 99; }
 [ "$GIT_REV" != "" ] || { echo "Could not determine git revision, aborting"; usage; exit 99; }
-[ "$NAME" != "" ] || { echo "-n name must be set"; usage; exit 99; }
+[ "$INPUT_NAME" != "" ] || { echo "-n name must be set"; usage; exit 99; }
 
 PULLED="-1"
 
-#Convert dashes to underscores to get tag names, except for validator, which for some reason is "e2e"
-tag_name=`echo $NAME | sed 's/-/_/g'`
-if [ $NAME == "validator" ]; then
-  tag_name="e2e";
+#Compute the tag name generated by the docker files.
+LOCAL_TAG=`echo $INPUT_NAME | sed 's/-/_/g'`
+LOCAL_TAG=libra_${LOCAL_TAG}
+
+# BEGIN DEALING WITH INCONSISTENT NAMING
+if [ $LOCAL_TAG == "libra_validator_tcb" ]; then
```

Comment:
We should open an issue to clean up the tagging scheme with the final one we will go with. cc @mgorven